source/synthDrivers/_espeak.py

```diff
@@ -123,23 +126,38 @@ def __eq__(self, other):
 @t_espeak_callback
 def callback(wav,numsamples,event):
 	try:
-		global player, isSpeaking, lastIndex
+		global player, isSpeaking, _numBytesPushed
 		if not isSpeaking:
 			return 1
+		indexes = []
 		for e in event:
 			if e.type==espeakEVENT_MARK:
-				lastIndex=int(e.id.name)
+				indexNum = int(e.id.name)
+				# e.audio_position is ms since the start of this utterance.
+				# Convert to bytes since the start of the utterance.
+				# samplesPerSec * 2 bytes per sample / 1000 ms per sec gives us bytes per ms.
+				indexByte = e.audio_position * player.samplesPerSec * 2 / 1000
+				# Subtract bytes in the utterance that have already been handled
+				# to give us the byte offset into the samples for this callback.
+				indexByte -= _numBytesPushed
+				indexes.append((indexNum, indexByte))
 			elif e.type==espeakEVENT_LIST_TERMINATED:
 				break
 		if not wav:
 			player.idle()
+			onIndexReached(None)
 			isSpeaking = False
 			return 0

```

Comment:
What does return 0 and return 1 mean? I am guessing they are NOT_SPEAKING and SPEAKING respectively. Could you change this to use a named constant?